packages/client/src/types.ts

```diff
@@ -163,6 +163,12 @@ export interface ConnectionLike {
    * @see https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal
    */
   withAbortSignal<R>(abortSignal: AbortSignal, fn: () => Promise<R>): Promise<R>;
+
+  /**
+   * Capability flag that determines whether the connection supports eager workflow start.
+   * This will only be true if the underlying connection is a {@link NativeConnection}.
+   */
+  readonly supportsEagerStart?: boolean;
```

Comment:
@THardy98 I did [a similar thing](https://github.com/temporalio/sdk-typescript/pull/1769/files#diff-aebd6517d8da25ba56bc895d268847cf7cc26a16275652f20ce3667438d4de61R371-R373) as part of #1769.

Not sure if there's value in exposing this one publicly.